### ISSUE TITLE
[Bug] Context menu items pop in and out of tabs' right click menu.

### ISSUE THREAD

**Issue by MagicalDrizzle** [piroor/treestyletab#3376]
## Abstract
Context menu entries pops in and out ("lag behind") of tabs' right click menu. This is because the list of right-click entries for each "type" of tab (parent, child, about:home) is slightly different. So right clicking a tab will cause its entries to show up, then right clicking on a different tab will initially show the previous tab's entries, which will quickly change to the current tab's entries.
## Steps to reproduce

Reproduced with both Firefox Stable and Firefox Nightly - and with TST on AMO and the nightly version of TST.
 1. Start Firefox with clean profile.
 2. Install TST.
 3. Open a tree as described below:
```
A (webpage)
\- B (webpage)
C (webpage)
D (about:home)
```
4. Right click the tabs. Make sure every right click is on a different tab to the last one (e.g A to C to B to D to C etc).

## Expected result
The right click menu and its entries should show up normally.

## Actual result
The right click menu seems to "lag behind" - Right clicking tab A then tab B, will cause tab A's entries to change to match tab B.
For example: On the left is the context menu entries of an empty tab, while on the right is of a parent tab with children.
![empty](https://github.com/piroor/treestyletab/assets/49761211/f5669134-bf86-41cd-a04d-e06049ebe663)![parent](https://github.com/piroor/treestyletab/assets/49761211/1e868c82-4f06-4540-b0d0-afd9c9f6c569)
The difference is the right tab's entries has another item: `Send this tree to device`, and `Send tab to device` is no longer grayed out.
Thus, right clicking the left tab then the right tab will cause the left tab's context entries to show up on the right tab for a moment, then `Send this tree to device` will suddenly show up and `Send tab to device` lights up.

https://github.com/piroor/treestyletab/assets/49761211/6c66cb79-8f5c-4f0a-bfcb-f7de93a76a18

## Environment

 * Platform (OS): Windows 10, Linux Mint
 * Version of Firefox:  118.0a1 nightly, 115.0.2 stable.
 * Version (or revision) of Tree Style Tab:  3.9.16

**Comment by piroor** [comment]
I can't confirm this problem on my environment but I know there is a risk of the context menu initialization: the status of menu items can be mixed if you reopen the menu immediately while asynchronous operations are in progress. I've introduced a safe guard mechanism with the commit 5cd0f2e so it should fix the problem.

**Comment by MagicalDrizzle** [comment]
I just tried the latest trunk build from Actions and unfortunately the problem is still there...

**Comment by piroor** [comment]
Hmm...

**Comment by piroor** [comment]
Ah! I've realized that I misread this issue - you didn't mean "wrong state context menu is shown", instead you meant "the context menu is initialized with delay". I've introduced more changes 9fd644a and b470840: now TST tries to update tab context menu before the menu is actually shown, if possible. You'll still see the delayed update if you reopen context menu very quickly, but visual flicking may go away on most cases if the context menu is shown with mouseup on your environment.
